src/txSerializers.ts

```diff
@@ -26,11 +26,40 @@ const serializeAmount = (amount: Amount) => {
     }
 }
 
-const serializeTxOutput = (output: TransactionOutput) =>
+const serializeLegacyTxOutput = (output: LegacyTransactionOutput) =>
     output.datumHash
-        ? [output.address, serializeAmount(output.amount), output.datumHash]
+        ? [output.address, serializeAmount(output.amount), output.datumHash.hash]
         : [output.address, serializeAmount(output.amount)]
 
+const serializeDatum = (datum: Datum) => {
+    switch (datum.type) {
+    case DatumType.HASH:
+        return [datum.type, datum.hash]
+    case DatumType.INLINE:
+        return [datum.type, new Tagged(CborTag.ENCODED_CBOR, datum.bytes)]
```

Comment:
Since HW wallets will receive only the datum contents without the tag and bytes envelope, I suggest interop-lib returns the unwrapped data after parsing (and conversely, adds the tag when serializing). This is implemented in https://github.com/vacuumlabs/cardano-hw-interop-lib/pull/7/commits/dc225b45e2b8765ddb3a3439b685067e904ca14b.